examples/backends/low_level/horqrux_backend.py

```diff
@@ -0,0 +1,67 @@
+from __future__ import annotations
+
+import jax.numpy as jnp
+import optax
+from jax import Array, jit, value_and_grad
+
+from qadence.backends import backend_factory
+from qadence.circuit import QuantumCircuit
+from qadence.constructors import feature_map, hea, total_magnetization
+
+backend = "horqrux"
+
+num_epochs = 10
+n_qubits = 4
+
+fm = feature_map(n_qubits)
+circ = QuantumCircuit(n_qubits, hea(n_qubits, depth=1))
+obs = total_magnetization(n_qubits)
+
+for diff_mode in ["ad", "gpsr"]:
+    hq_bknd = backend_factory(backend, diff_mode)
+    hq_circ, hq_obs, hq_embedfn, hq_init_params = hq_bknd.convert(circ, obs)
+    inputs = {}
+    embedded_params = hq_embedfn(hq_init_params, inputs)
+    optimizer = optax.adam(learning_rate=0.001)
+
+    param_names = embedded_params.keys()
+    param_values = embedded_params.values()
+    init_array = jnp.array(jnp.concatenate([arr for arr in param_values]))
+    opt_state = optimizer.init(init_array)
+    param_array = init_array
```

Comment:
This is all not necessary in Jax... we can just give the embedded params directly to the optimiser, no need to unpack.

Reply:
jep, changed